jbmc/src/java_bytecode/assignments_from_json.cpp

```diff
@@ -199,9 +199,13 @@ static bool has_nondet_length(const jsont &json)
 /// (typed vs. untyped).
 static jsont get_untyped(const jsont &json, const std::string &object_key)
 {
-  if(get_type(json) || has_nondet_length(json))
-    return json[object_key];
-  return json;
+  if(!json.is_object())
+    return json;
+
+  const auto &json_object = to_json_object(json);
+  PRECONDITION(
+    get_type(json) || json_object.find("@nondetLength") != json_object.end());
```

Comment:
:pick: We could be more precise and assert the current precondition only if `json` is a `json_arrayt`, and assert only `get_type(json)` otherwise, as we wouldn't expect a `@nondetLength` key on e.g. a number or boolean. But then again in the current version we would just ignore such `@nondetLength` keys, which wouldn't cause any harm.

Reply:
precondition is split

Reply:
This introduced problems, so I reverted to the initial solution.